pkg/images/enricher/enricher_v2_impl.go

```diff
@@ -0,0 +1,1035 @@
+package enricher
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/delegatedregistry"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/images"
+	"github.com/stackrox/rox/pkg/images/cache"
+	"github.com/stackrox/rox/pkg/images/integration"
+	"github.com/stackrox/rox/pkg/images/utils"
+	"github.com/stackrox/rox/pkg/integrationhealth"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/protoutils"
+	registryTypes "github.com/stackrox/rox/pkg/registries/types"
+	"github.com/stackrox/rox/pkg/sac"
+	scannerTypes "github.com/stackrox/rox/pkg/scanners/types"
+	"github.com/stackrox/rox/pkg/signatures"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
+	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"golang.org/x/time/rate"
+)
+
+var _ ImageEnricherV2 = (*enricherV2Impl)(nil)
+
+type enricherV2Impl struct {
+	cvesSuppressor CVESuppressor
+	integrations   integration.Set
+
+	errorsPerRegistry  map[registryTypes.ImageRegistry]int32
+	registryErrorsLock sync.RWMutex
+	errorsPerScanner   map[scannerTypes.ImageScannerWithDataSource]int32
+	scannerErrorsLock  sync.RWMutex
+
+	integrationHealthReporter integrationhealth.Reporter
+
+	metadataLimiter *rate.Limiter
+	metadataCache   cache.ImageMetadata
```

Comment:
Out of scope of the PR, but something to think about in another ticket.  Is the flattening going to increase the memory needed for the cache?